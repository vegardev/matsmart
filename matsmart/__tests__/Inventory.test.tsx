import React from "react";
import { render, waitFor } from "@testing-library/react";
import Freezer from "@/src/app/inventory/freezer/page";
import Fridge from "@/src/app/inventory/fridge/page";
import Pantry from "@/src/app/inventory/pantry/page";
import {
  freezerInventoryDummyData,
  fridgeInventoryDummyData,
  pantryInventoryDummyData,
} from "@/src/app/backend/dummyData";

describe("Freezer", () => {
  it("renders correctly", async () => {
    fetch.mockResponseOnce(JSON.stringify(freezerInventoryDummyData));

    const { asFragment } = render(<Freezer />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});

describe("Fridge", () => {
  it("renders correctly", async () => {
    fetch.mockResponseOnce(JSON.stringify(fridgeInventoryDummyData));

    const { asFragment } = render(<Fridge />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});

describe("Pantry", () => {
  it("renders correctly", async () => {
    fetch.mockResponseOnce(JSON.stringify(pantryInventoryDummyData));

    const { asFragment } = render(<Pantry />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
