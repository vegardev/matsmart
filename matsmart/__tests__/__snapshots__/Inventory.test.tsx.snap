// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryPage renders Freezer correctly 1`] = `
<DocumentFragment>
  <h1>
    Fryser
  </h1>
  <table
    class="min-w-full divide-y divide-gray-200"
  >
    <thead
      class="bg-gray-50"
    >
      <tr>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Groceries
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Quantity
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Expiration Date
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Remove?
        </th>
      </tr>
    </thead>
    <tbody
      class="bg-white divide-y divide-gray-200"
    />
  </table>
</DocumentFragment>
`;

exports[`InventoryPage renders Fridge correctly 1`] = `
<DocumentFragment>
  <h1>
    Kj√∏leskap
  </h1>
  <table
    class="min-w-full divide-y divide-gray-200"
  >
    <thead
      class="bg-gray-50"
    >
      <tr>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Groceries
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Quantity
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Expiration Date
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Remove?
        </th>
      </tr>
    </thead>
    <tbody
      class="bg-white divide-y divide-gray-200"
    />
  </table>
</DocumentFragment>
`;

exports[`InventoryPage renders Pantry correctly 1`] = `
<DocumentFragment>
  <h1>
    Skuff
  </h1>
  <table
    class="min-w-full divide-y divide-gray-200"
  >
    <thead
      class="bg-gray-50"
    >
      <tr>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Groceries
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Quantity
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Expiration Date
        </th>
        <th
          class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
        >
          Remove?
        </th>
      </tr>
    </thead>
    <tbody
      class="bg-white divide-y divide-gray-200"
    />
  </table>
</DocumentFragment>
`;
